### ISSUE TITLE
Fast scan mode

### ISSUE THREAD

**Issue by tesshucom** [tesshucom/jpsonic#1101]
#### Overview

Added options to the settings screen to control checking for update dates during scanning. There are advantages such as faster scanning and reduced load on the server machine. And it's also a condition for Google Drive support.

____

#### Web page

``Music folders settings > Other operations > File change check method``

Here you can choose between the traditional scan check method and the new simplified check method.

![image](https://user-images.githubusercontent.com/27724847/128634540-5da81249-2686-4b45-904d-e7de0494e851.png)

"Manual management by explicit specification" can avoid update date access on Java, which tends to be a bottleneck on HDD-configured (not SSD) machines. If you select this option, "Specify the album to be scanned" is inevitably enabled. Users can enable the next scan with a new button on the album page.

![image](https://user-images.githubusercontent.com/27724847/128634669-8f4e0cae-e189-44de-b904-b1adf2939950.png)

 - "Specify the album to be scanned" can be used in combination with the conventional scanning method.
 - Ignore timestamps is a hidden option on legacy servers, an option had used only by those who know it. Useful scenes are limited.

<details>
<summary>details</summary>

With the addition of the #1082 feature, it is possible to reproduce a manual scan, that is different from the conventional automatic scan based on the update date.

 - Traditional scanning methods are very verbose, but they are basic logic and cannot be deleted. It is essential in situations where the user cannot completely predict data updates, for example scanning the Postcast directory or scanning the mirrored directory.
 - On the other hand, it is a logic that depends on Java's lastModified, and has a structure that is destined to become a bottleneck. But, It's difficult to avoid this problem while maintaining cross-platform.
 - Multithreading is one solution, but the number of files that can be polled depends on the OS, so that alone cannot be a universal final solution. (Multithreading  serves as one of several measures to be taken, but it is not the wedge of top priority.)
 - We can suggest scans that use only LastScanned without LastModified. 
 - With this method, new creation and deletion of music data is the same as before, but when updating tags etc., the user needs to set a "flag" on the album page. Only flagged albums will be updated in the next scan.
 - This method should be considered as limited logic, not as an alternative logic to the traditional scanning method. In other words, users should be able to understand and use the differences between each scanning method, and should be able to switch at any time. 

Scanning that relies solely on LastScanned is advantageous when the music folder is completely under the control of the administrator. It is not suitable for scanning directories that are mirrored, but otherwise it will speed up the scan because bottlenecks can be avoided. For example, if you have a library of tens of thousands of songs and you want to add a few albums a month or edit tags, it can be a beneficial feature for (most) users. In addition, the method that does not rely on LastModified is also advantageous for scanning in an environment where LastModified does not work as expected (Google Drive, etc.).

- The speed of the first scan is the same as before. (It's the cost to be paid.)
- A big change is the determination of whether to scan a scanned file. In other words, improvement can be expected for the second and subsequent scans, and the scan execution time can be reduced to about half at the maximum. 
- This effect is an assumption for NAS using relatively old hard disks that can exist in the home. In a state-of-the-art environment with SSDs, the difference may be subtle. Or even in a state-of-the-art environment equipped with SSD, the difference may become noticeable depending on the communication capacity of the network. 
- In other words, this improvement can be expected to be more effective in a weak environment.

</details>

**Comment by tesshucom** [comment]
~~Experimental : Currently the focus is on the update flow. In the official release, add / remove will be supported and it will be available as a more convenient scanning method.~~

Fixed in #1108.
